pkg/monitortests/kubeapiserver/auditloganalyzer/monitortest/clustermonitor.go

```diff
@@ -0,0 +1,127 @@
+package monitortest
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/openshift/origin/pkg/monitor/monitorapi"
+	"github.com/openshift/origin/pkg/monitortestframework"
+	"github.com/openshift/origin/pkg/monitortests/kubeapiserver/auditloganalyzer"
+	"github.com/openshift/origin/pkg/monitortests/kubeapiserver/auditloganalyzer/eventsprovider"
+	"github.com/openshift/origin/pkg/monitortests/kubeapiserver/auditloganalyzer/summary"
+	"github.com/openshift/origin/pkg/test/ginkgo/junitapi"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+// NewAuditLogAnalyzer returns a cluster monitoring test that analyzes the
+// cluster audit log events.
+func NewAuditLogAnalyzer() monitortestframework.MonitorTest {
+	return &clusterMonitor{
+		collectors: []auditloganalyzer.AuditEventCollector{
+			summary.NewAnalyzer(),
+		},
+	}
+}
+
+type clusterMonitor struct {
+	config     *rest.Config
+	collectors []auditloganalyzer.AuditEventCollector
```

Comment:
rather than have collectors that this monitor test doesn't know the meaning of, if you have individual collectors that you **do** know the meaning of, you can appropriately gather intervals and junits and serialization directly inline rather than create a secondary framework to chase them around